libslide/src/bignum.rs

```diff
@@ -24,6 +26,11 @@ impl Bignum {
         let mut has_decimal = false;
         let mut int = Vec::new();
         let mut dec = Vec::new();
+
+        if input.is_empty() {
+            panic!("A arbitray precision number must have a valid input");
+        }
+
```

Comment:
Maybe it's better to return an option, but we can do that in a different pr

Reply:
how would we retunr an option in the constructor

Reply:
Just change the return type of the method